client/containers/DashboardCollectionLayout/DashboardCollectionLayout.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { useBeforeUnload, useUpdateEffect } from 'react-use';
+import { AnchorButton, Button, ButtonGroup, RadioGroup, Radio } from '@blueprintjs/core';
+
+import {
+	DashboardFrame,
+	SettingsSection,
+	LayoutEditor,
+	LinkedPageSelect,
+	InputField,
+} from 'components';
+import { usePageContext } from 'utils/hooks';
+import { Collection, Pub } from 'utils/types';
+import * as api from 'client/utils/collections/api';
+
+require('./dashboardCollectionLayout.scss');
+
+type Props = {
+	collection: Collection;
+	pubs: Pub[];
+};
+
+const DashboardCollectionLayout = (props: Props) => {
+	const { communityData, scopeData } = usePageContext();
+	const { activeCommunity } = scopeData.elements;
+	const [persistedCollectionData, setPersistedCollectionData] = useState(props.collection);
+	const [pendingCollectionData, setPendingCollectionData] = useState<Partial<Collection>>({});
+	const [isPersisting, setIsPersisting] = useState(false);
+	const collection = { ...persistedCollectionData, ...pendingCollectionData };
+	const [isUsingBlocks, setIsUsingBlocks] = useState(!collection.pageId);
+	const { layout = { blocks: [] } } = collection;
+	const hasPendingChanges = Object.keys(pendingCollectionData).length > 0;
+	const canPersistChanges = hasPendingChanges && (isUsingBlocks || collection.pageId);
+
+	const handleSaveChanges = useCallback(async () => {
+		setIsPersisting(true);
+		await api.updateCollection({
+			communityId: communityData.id,
+			collectionId: collection.id,
+			updatedCollection: pendingCollectionData,
+		});
+		setIsPersisting(false);
+		setPersistedCollectionData((data) => ({ ...data, ...pendingCollectionData }));
+		setPendingCollectionData({});
+	}, [collection.id, communityData.id, pendingCollectionData]);
+
+	const updateCollection = useCallback(
+		(update: Partial<Collection>) =>
+			setPendingCollectionData((data) => ({ ...data, ...update })),
+		[],
+	);
+
+	const updateLayout = useCallback(
+		(update: Partial<Collection['layout']>) =>
+			setPendingCollectionData((data) => ({ ...data, layout: { ...layout, ...update } })),
+		[layout],
+	);
+
+	useUpdateEffect(() => {
+		if (isUsingBlocks) {
+			updateCollection({ pageId: null });
+		}
+	}, [isUsingBlocks, updateCollection]);
+
+	useBeforeUnload(
+		hasPendingChanges,
+		'You have unsaved changes to this Collection. Are you sure you want to navigate away?',
+	);
+
+	const renderControls = () => {
+		return (
+			<React.Fragment>
+				<AnchorButton icon="share" href={`/${collection.slug}`}>
+					Visit Collection
+				</AnchorButton>
+				<Button
+					type="button"
+					intent="primary"
+					text="Save Changes"
+					disabled={!canPersistChanges}
+					loading={isPersisting}
+					onClick={handleSaveChanges}
+				/>
+			</React.Fragment>
+		);
+	};
+
+	const renderPageSelectLabel = () => {
+		return (
+			<>
+				Use a Page as the landing page for this Collection: &nbsp;
```

Comment:
Would `&nbsp;` have the same effect as `{' '}`? Do we prefer one over the other?

Reply:
I wish I could say why, but I only know that `&nbsp;` works here while `{' '}` is not treated as visible whitespace (maybe because it abuts an opening tag?)